### ISSUE TITLE
Possible Mismatch Between README Description and 'privInfo' Usage in train_student_sim.sh

### ISSUE THREAD

**Issue by DFREANDY** [HaozhiQi/penspin#1]
### Description: 
Hello,
I am a master's student in robotics at SJTU, and I am following your work on dexterous hand manipulation. 
Thank you for your open-source project!!! It helped me learn a lot about dexterous hand reinforcement learning from scratch. 
However, I have the following doubts：

In the README, `Step 2: Student Policy Pretraining` mentions that we train a proprioceptive student policy by distilling from the trained oracle policy. However, when I look at the configuration in `train_student_sim.sh`, it seems that certain privInfo options are enabled, which might contradict the goal of training a proprioceptive student policy.

The [train_student_sim.sh](https://github.com/HaozhiQi/penspin/blob/main/scripts/train_student_sim.sh) script includes the following configurations:
```
task.env.privInfo.enable_obj_orientation=True 
task.env.privInfo.enable_ft_pos=True 
task.env.privInfo.enable_obj_angvel=True 
task.env.privInfo.enable_tactile=True 
```
These configurations enable various privInfo options such as object orientation, object angular velocity, fingertip position, and tactile information, which seem to go beyond the proprioceptive data and might introduce additional sensory information (e.g., visual or tactile) not directly related to proprioception.

### Expected Behavior: 
From the README description, I expected the student policy to be trained purely with proprioceptive data, without the inclusion of additional information like object orientation or tactile data. Therefore, these privInfo configurations in the script seem inconsistent with the goal of training a proprioceptive student policy.

### Actual Behavior: 
It appears that the current configuration in train_student_sim.sh includes options that are not strictly related to proprioception, potentially affecting the training process.

**Comment by HaozhiQi** [comment]
Hi @DFREANDY, I apologize for the confusion.

In the current implementation, the student policy training is done in [this class](https://github.com/HaozhiQi/penspin/blob/f6f2c87ed6a427edd8929751e8feb4bc1744b197/penspin/algo/ppo/demon.py). (see `train.algo=DemonTrain` in the script).

In this class, `self.demon_model` is the teacher policy and `self.model` is the student policy. We set `student: True` for `self.model`. In this case, it will force the `self.model` to [not use priv info](https://github.com/HaozhiQi/penspin/blob/f6f2c87ed6a427edd8929751e8feb4bc1744b197/penspin/algo/models/models.py#L232).

Sorry I realize this is a bit counterintuitive but I have not got a chance to refactor the codebase. Let me know if you have further questions!

**Comment by DFREANDY** [comment]
Thank you for your reply! @HaozhiQi 
Now I understand the logic of this part of your code. And I have a new question: Does `fingertip_pose `belong to `priv_info`?

I think this information can be obtained either through forward kinematics or external perception. In the code, it seems to be directly retrieved from the simulation environment. And it belongs to `priv_info`. However, when training the student policy, the observation includes [`fingertip_pose`](https://github.com/HaozhiQi/penspin/blob/f6f2c87ed6a427edd8929751e8feb4bc1744b197/penspin/algo/models/models.py#L243).

During the `Step 1` training process, I also found that adding fingertip-related reward terms (e.g., [`ft_angvel_at_cf`](https://github.com/HaozhiQi/penspin/blob/f6f2c87ed6a427edd8929751e8feb4bc1744b197/penspin/tasks/allegro_hand_hora.py#L540)) to the reward function had a significant impact on policy training, improving its performance.

Have you observed similar effects or gained any insights from your own policy training experiments?

Thank you again for maintaining this project. Looking forward to your further response!

**Comment by HaozhiQi** [comment]
Hi @DFREANDY, yes we did use the fingertip pose in the student and final policy. The performance improvements of adding fingertip pose is not very significant though. Sorry for the confusion, we should update the paper.

More details: in student distillation, it was used [here](https://github.com/HaozhiQi/penspin/blob/main/penspin/algo/models/models.py#L244). The fingertip is randomized to make it robust to real-world observations. In the real world, the fingertip pose can be calculated from joint positions because we know the hand's URDF. The code is [here](https://github.com/HaozhiQi/penspin/blob/main/real/robot_controller/real_robot/robot.py#L488).

Regarding `ft_angvel_at_cf`, I don't think we explored adding this. It's interesting that it significantly improves the policy, probably because it contains explicit dynamics information on the fingertip, which is closely related to the contact between the fingertip and the object.

**Comment by DFREANDY** [comment]
Thank you for your sincere response! @HaozhiQi  It has resolved my doubt.🫡

Actually, I am currently trying to reproduce this work on the Leap Hand, and I think that I may encounter more issues in the future. I hope to continue consulting with you! ☺️🤝

I will also share my findings and experiences during the reproduction process, in order to help others who are attempting to replicate this work.

**Comment by HaozhiQi** [comment]
@DFREANDY, I'm looking forward to seeing your LEAP hand results! And also thank you for sharing your findings. 

I'll close this issue for now, but Feel free to open new issues for any other specific questions you encounter, I'll try my best to reply as quick as possible. Good luck with your reproduction process!
